sql/core/src/main/scala/org/apache/spark/sql/execution/subquery.scala

```diff
@@ -191,7 +194,10 @@ case class PlanSubqueries(sparkSession: SparkSession) extends Rule[SparkPlan] {
           )
         }
         val executedPlan = QueryExecution.prepareExecutedPlan(sparkSession, query)
-        InSubqueryExec(expr, SubqueryExec(s"subquery#${exprId.id}", executedPlan), exprId)
+        InSubqueryExec(
+          expr,
+          SubqueryExec(s"subquery#${exprId.id}", executedPlan, maxNumRows = None),
+          exprId)
```

Comment:
So, we need this change because we don't prefer to have the default value for `maxNumRows`?